design-proposals/primary_network_binding_for_seamless_migration.md

```diff
@@ -0,0 +1,596 @@
+  maintenance. E.g. the logic and codebase should not diverge too much
+  from the existing core network flow.
+
+### Pre Requirements
+
+The main challenge when working with the primary pod network is to
+assure connectivity recovers after a VM migration. A new requirement is
+added with this proposal, to preserve existing connectivity (e.g. TCP
+connection) with minimal (to none) disruption.
+
+As we are focusing here on the primary pod network, these requirements
+translate to several pre-requirements from the network provider & CNI:
+
+- Sticky IP: An IPAM plugin assigns an IP for each pod. In order to
+  support migration and preserve seamless connectivity, the pod IP on
+  the target node must use the same IP.
+  The network-provider/CNI/IPAM needs to assure such an option exists so
+  the same IP used for the primary pod network can be moved to the
+  target pod.
+- Flat Cluster Subnet: In order to allow the same IP to be assigned on a
+  different node when migration occurs, only a single subnet needs to
+  exist from the entire cluster on the primary pod network.
+- Traffic hand-off on migration: With the need to keep the same IP on
+  both the source and target nodes/pods, the network-provider needs to
+  assure traffic is correctly directed to the active VM. I.e.during the
+  migration process, the source VM is the active one (which may take
+  several tens of seconds), afterward when the hand-off is moved to the
+  target, traffic should start being directed to the new location.
+  A Kubevirt VM signals that it is now active by sending a GARP message,
+  letting all the bridges on the network know that the mac address
+  moved. This may be enough for some network-providers while others may
+  use other techniques to correctly forward the traffic to the right
+  port.
+
+> **Note**: In case this is not handled properly, traffic may be disrupted
+> during and following a migration in unexpected ways.
+
+These are the common pre-requirements for all the solutions options that
+this proposal includes. Each proposed binding may have unique
+requirements of their own.
+
+### Network Binding Options
+
+There are a few existing bindings (passt, bridge) and non-existing
+futuristic ones which can be developed to solve the needs. This section
+presents each with their pros and cons.
+
+The chosen solution and a table summary is provided first, followed by a
+detailed description of each.
+
+<!-- TOC -->
+* [The Binding Winner](#the-binding-winner)
+* [Summary](#summary-1)
+* [Passt Binding](#passt-binding)
+* [Bridge Binding](#bridge-binding)
+  * [As a Binding Plugin](#as-a-binding-plugin)
+  * [As a DomainAttachmentType](#as-a-bridge-domainattachmenttype)
+* [Macvtap Binding](#macvtap-binding)
+  * [As a DomainAttachmentType](#as-a-macvtap-domainattachmenttype)
+* [Other L2 Bindings](#other-l2-bindings)
+<!-- TOC -->
+
+#### The Binding Winner
+
+The bridge binding plugin option has been chosen after a successful
```

Comment:
It wasn't clear to me that the `As a bridge DomainAttachmentType` is the winner.

Reply:
Done